File: src/main/java/ru/hollowhorizon/mastertech/util/SpawnHelper.java
package ru.hollowhorizon.mastertech.util;

import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.WorldServer;

public class SpawnHelper {
    public static void teleportToDim(EntityPlayerMP sp, int dimension) {
        WorldServer serverWorld = sp.getServerWorld();

        if (!serverWorld.isRemote)
            sp.changeDimension(
                    dimension,
                    (world, entity, yaw) -> {}
            );
    }
}

File: src/main/java/ru/hollowhorizon/mastertech/util/SpawnHelper.kt
package ru.hollowhorizon.mastertech.util

import net.minecraft.entity.player.EntityPlayerMP

object SpawnHelper {
    @JvmStatic
    fun teleportToDim(sp: EntityPlayerMP, dimension: Int) {
        val serverWorld = sp.serverWorld

        if (!serverWorld.isRemote) sp.changeDimension(
            dimension
        ) { world, entity, yaw -> }
    }
}
